lib/utils/display.js

```diff
@@ -1,174 +1,207 @@
-  }
+  // progress
+  #progress = null
 
-  static clean (output) {
-    if (typeof output === 'string') {
-      // Strings are cleaned inline
-      return stripC01(output)
-    }
-    if (!output || typeof output !== 'object') {
-      // Numbers, booleans, null all end up here and don't need cleaning
-      return output
-    }
-    // output && typeof output === 'object'
-    // We can't use hasOwn et al for detecting the original but we can use it
-    // for detecting the properties we set via defineProperty
-    if (
-      output[inspect.custom] &&
-      (!Object.hasOwn(output, originalCustomInspect))
-    ) {
-      // Save the old one if we didn't already do it.
-      Object.defineProperty(output, originalCustomInspect, {
-        value: output[inspect.custom],
-        writable: true,
-      })
-    }
-    if (!Object.hasOwn(output, originalCustomInspect)) {
-      // Put a dummy one in for when we run multiple times on the same object
-      Object.defineProperty(output, originalCustomInspect, {
-        value: function () {
-          return this
-        },
-        writable: true,
-      })
-    }
-    // Set the custom inspect to our own function
-    Object.defineProperty(output, inspect.custom, {
-      value: function () {
-        const toClean = this[originalCustomInspect]()
-        // Custom inspect can return things other than objects, check type again
-        if (typeof toClean === 'string') {
-          // Strings are cleaned inline
-          return stripC01(toClean)
-        }
-        if (!toClean || typeof toClean !== 'object') {
-          // Numbers, booleans, null all end up here and don't need cleaning
-          return toClean
-        }
-        return stripC01(inspect(toClean, { customInspect: false }))
-      },
-      writable: true,
-    })
-    return output
-  }
+  // options
+  #levelIndex = 0
+  #timing = false
+  #json = false
+  #heading = 'npm'
+
+  #stdout = null
+  #stderr = null
 
-  on () {
+  constructor ({ stdout, stderr }) {
+    this.#stdout = stdout
+    this.#stderr = stderr
     process.on('log', this.#logHandler)
   }
 
   off () {
     process.off('log', this.#logHandler)
-    // Unbalanced calls to enable/disable progress
-    // will leave change listeners on the tracker
-    // This pretty much only happens in tests but
-    // this removes the event emitter listener warnings
-    log.tracker.removeAllListeners()
+    this.#logBuffer.length = 0
+    if (this.#progress) {
+      this.#progress.stop()
+    }
   }
 
-  load (config) {
-    const {
-      color,
-      chalk,
-      timing,
-      loglevel,
-      unicode,
-      progress,
-      silent,
-      heading = 'npm',
-    } = config
-
+  load ({ loglevel, chalk, timing, unicode, progress, json, heading }) {
     this.#chalk = chalk
+    this.#colors = COLOR_PALETTE({ chalk })
+
+    this.#levelIndex = LEVEL_OPTIONS[loglevel].index
+    this.#timing = timing
+    this.#json = json
+    this.#heading = heading
```

Comment:
Why have defaults on these if we set them during load? Is it possible to do anything before we've called load?  If so then why do we have load, if not why isn't this part of the constructor?

Reply:
Removed the defaults on these. That's just a habit so that when the private property is initiated I can see what type it is, but it's not necessary and more likely to create bugs since the real values come from config which also has defaults.